In [1]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

##### Load your dataset into python using the pandas library and save data into a dataframe named dfi (where i is one of your datasets, for a total of 4).  

In [9]:
# importing datasets
# weather URL from GitHub (source: https://github.com/data602sps/datasetspractice)
weather_url = "https://raw.githubusercontent.com/data602sps/datasetspractice/main/weather.csv"
# df1 : weather data
df1 = pd.read_csv(weather_url)

# salary URL from GitHub (source: https://github.com/data602sps/datasetspractice)
salary_url = "https://raw.githubusercontent.com/data602sps/datasetspractice/main/salarydata.csv"
# df2 : salary data
df2 = pd.read_csv(salary_url)

# birds URL from data.gov (source: https://catalog.data.gov/dataset/riparian-birds-sierra-nevada-foothills-ds303-3e51a)
birds_url = "https://data-cdfw.opendata.arcgis.com/datasets/e1b1ff8f4872409eb43447175105e8ce_0.csv?outSR=%7B%22latestWkid%22%3A3857%2C%22wkid%22%3A102100%7D"
# df3 : birds data
df3 = pd.read_csv(birds_url)

# tate artist data from GitHub (source: https://github.com/tategallery/collection/blob/master/artist_data.csv)
tate_url = "https://raw.githubusercontent.com/tategallery/collection/master/artist_data.csv"
# df4 : tate data
df4 = pd.read_csv(tate_url)

##### Preview your data by calling your dataframe’s name. How many columns and rows do you see?  

In [15]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8719 entries, 0 to 8718
Data columns (total 15 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  8719 non-null   int64  
 1   origin      8719 non-null   object 
 2   year        8719 non-null   int64  
 3   month       8718 non-null   float64
 4   day         8718 non-null   float64
 5   hour        8718 non-null   float64
 6   temp        8718 non-null   float64
 7   dewp        8718 non-null   float64
 8   humid       8718 non-null   float64
 9   wind_dir    8486 non-null   float64
 10  wind_speed  8718 non-null   float64
 11  wind_gust   8718 non-null   float64
 12  precip      8719 non-null   float64
 13  pressure    7780 non-null   float64
 14  visib       8719 non-null   float64
dtypes: float64(12), int64(2), object(1)
memory usage: 1021.9+ KB


df1: 8719 rows, 14 columns

In [16]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 51 entries, 0 to 50
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Sate                   51 non-null     object 
 1   2018 Mean Annual Wage  51 non-null     object 
 2   Unnamed: 2             0 non-null      float64
dtypes: float64(1), object(2)
memory usage: 1.3+ KB


df2 : 51 rows, 3 columns

In [17]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 144 entries, 0 to 143
Data columns (total 8 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   X         144 non-null    float64
 1   Y         144 non-null    float64
 2   OBJECTID  144 non-null    int64  
 3   SEGMENT   144 non-null    int64  
 4   POINT     144 non-null    int64  
 5   LATDD     144 non-null    float64
 6   LONDD     144 non-null    float64
 7   SEGPTID   144 non-null    int64  
dtypes: float64(4), int64(4)
memory usage: 9.1 KB


df3 : 144 rows, 8 columns

In [18]:
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3532 entries, 0 to 3531
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            3532 non-null   int64  
 1   name          3532 non-null   object 
 2   gender        3416 non-null   object 
 3   dates         3470 non-null   object 
 4   yearOfBirth   3472 non-null   float64
 5   yearOfDeath   2228 non-null   float64
 6   placeOfBirth  3040 non-null   object 
 7   placeOfDeath  1453 non-null   object 
 8   url           3532 non-null   object 
dtypes: float64(2), int64(1), object(6)
memory usage: 248.5+ KB


df4 : 3532 rows, 8 columns

##### Examine the shape of your data using the .shape command and the data types of your columns using .dtypes.  

In [44]:
df1.shape

(8719, 15)

In [45]:
df1.dtypes

Unnamed: 0      int64
origin         object
year            int64
month         float64
day           float64
hour          float64
temp          float64
dewp          float64
humid         float64
wind_dir      float64
wind_speed    float64
wind_gust     float64
precip        float64
pressure      float64
visib         float64
dtype: object

In [46]:
df2.shape

(51, 3)

In [47]:
df2.dtypes

Sate                      object
2018 Mean Annual Wage     object
Unnamed: 2               float64
dtype: object

In [48]:
df3.shape

(144, 8)

In [49]:
df3.dtypes

X           float64
Y           float64
OBJECTID      int64
SEGMENT       int64
POINT         int64
LATDD       float64
LONDD       float64
SEGPTID       int64
dtype: object

In [50]:
df4.shape

(3532, 9)

In [51]:
df4.dtypes

id                int64
name             object
gender           object
dates            object
yearOfBirth     float64
yearOfDeath     float64
placeOfBirth     object
placeOfDeath     object
url              object
dtype: object

##### Use .describe() on your data. What do you notice about your data? What does this command return? 

In [53]:
df1.describe()

Unnamed: 0    year        month          day         hour  \
count  8719.000000  8719.0  8718.000000  8718.000000  8718.000000   
mean   4360.000000  2013.0     6.506538    15.679628    11.515715   
std    2517.102832     0.0     3.439961     8.768250     6.916846   
min       1.000000  2013.0     1.000000     1.000000     0.000000   
25%    2180.500000  2013.0     4.000000     8.000000     6.000000   
50%    4360.000000  2013.0     7.000000    16.000000    12.000000   
75%    6539.500000  2013.0     9.000000    23.000000    18.000000   
max    8719.000000  2013.0    12.000000    31.000000    23.000000   

              temp         dewp        humid     wind_dir   wind_speed  \
count  8718.000000  8718.000000  8718.000000  8486.000000  8718.000000   
mean     55.484942    41.794184    62.863333   193.194674     9.335732   
std      18.345817    19.619776    19.572699   110.541333    12.328261   
min      10.940000    -9.040000    13.950000     0.000000     0.000000   
25%      39.920000    26.060000    47.602500   110.000000     5.753900   
50%      55.940000    42.800000    62.330000   220.000000     9.206240   
75%      71.060000    57.920000    78.790000   280.000000    12.658580   
max     100.040000    75.920000   100.000000   360.000000  1048.360580   

         wind_gust       precip     pressure        visib  
count  8718.000000  8719.000000  7780.000000  8719.000000  
mean     10.743374     0.003035  1017.826465     9.224522  
std      14.187117     0.022686     7.396722     2.094044  
min       0.000000     0.000000   983.900000     0.120000  
25%       6.621473     0.000000  1012.800000    10.000000  
50%      10.594357     0.000000  1017.500000    10.000000  
75%      14.567241     0.000000  1022.900000    10.000000  
max    1206.432388     1.180000  1041.900000    10.000000

In [54]:
df2.describe()

Unnamed: 2
count         0.0
mean          NaN
std           NaN
min           NaN
25%           NaN
50%           NaN
75%           NaN
max           NaN

In [55]:
df3.describe()

X             Y    OBJECTID       SEGMENT       POINT  \
count  1.440000e+02  1.440000e+02  144.000000    144.000000  144.000000   
mean  -1.345920e+07  4.680023e+06   72.500000  37003.006944    2.590278   
std    3.234224e+04  6.231122e+04   41.713307  13112.299743    1.270500   
min   -1.353575e+07  4.592517e+06    1.000000  19574.000000    1.000000   
25%   -1.348335e+07  4.628830e+06   36.750000  26649.000000    1.750000   
50%   -1.345011e+07  4.674343e+06   72.500000  36727.000000    2.500000   
75%   -1.343586e+07  4.728761e+06  108.250000  37912.250000    3.250000   
max   -1.341135e+07  4.819991e+06  144.000000  62977.000000    5.000000   

            LATDD       LONDD        SEGPTID  
count  144.000000  144.000000     144.000000  
mean    38.707312 -120.906078  370032.659722  
std      0.436263    0.290535  131123.092625  
min     38.092609 -121.593667  195741.000000  
25%     38.348882 -121.122980  266491.750000  
50%     38.668814 -120.824340  367271.500000  
75%     39.049470 -120.696399  379125.750000  
max     39.683050 -120.476225  629774.000000

In [56]:
df4.describe()

id  yearOfBirth  yearOfDeath
count   3532.000000  3472.000000  2228.000000
mean    3330.331257  1887.001440  1921.356822
std     4160.172543    77.140335    79.520671
min        0.000000  1497.000000  1543.000000
25%      885.750000  1855.000000  1876.000000
50%     1772.500000  1910.000000  1944.000000
75%     2676.250000  1941.000000  1983.000000
max    18896.000000  2004.000000  2014.000000

.describe() provides summary statistics of the columns in the dataframe. For df2, there are no statistics because the data is not in numerical form. 

##### Use the .head() and .tail() command - what does this do?  

In [59]:
df1.head()

Unnamed: 0 origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0           1    EWR  2013    1.0  1.0   0.0  37.04  21.92  53.97     230.0   
1           2    EWR  2013    1.0  1.0   1.0  37.04  21.92  53.97     230.0   
2           3    EWR  2013    1.0  1.0   2.0  37.94  21.92  52.09     230.0   
3           4    EWR  2013    1.0  1.0   3.0  37.94  23.00  54.51     230.0   
4           5    EWR  2013    1.0  1.0   4.0  37.94  24.08  57.04     240.0   

   wind_speed  wind_gust  precip  pressure  visib  
0    10.35702  11.918651     0.0    1013.9   10.0  
1    13.80936  15.891535     0.0    1013.0   10.0  
2    12.65858  14.567241     0.0    1012.6   10.0  
3    13.80936  15.891535     0.0    1012.7   10.0  
4    14.96014  17.215830     0.0    1012.8   10.0

In [60]:
df1.tail()

Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
8714     200.0     4.60312   5.297178     0.0       NaN    4.0  
8715      60.0     4.60312   5.297178     0.0    1008.1   10.0  
8716      40.0     4.60312   5.297178     0.0    1008.5   10.0  
8717      40.0     4.60312   5.297178     0.0    1023.9   10.0  
8718     210.0     8.05546   9.270062     0.0    1011.9   10.0

In [61]:
df2.head()

Sate  2018 Mean Annual Wage  Unnamed: 2
0    Alabama                 90,620         NaN
1      Alaska                99,180         NaN
2     Arizona                88,800         NaN
3    Arkansas                81,430         NaN
4  California                97,110         NaN

In [62]:
df2.tail()

Sate  2018 Mean Annual Wage  Unnamed: 2
46       Virginia                91,700         NaN
47     Washington                85,930         NaN
48  West Virginia                89,420         NaN
49      Wisconsin                85,200         NaN
50        Wyomimg                87,510         NaN

In [63]:
df3.head()

X             Y  OBJECTID  SEGMENT  POINT      LATDD  \
0 -1.352692e+07  4.801859e+06         1    19574      1  39.557583   
1 -1.352698e+07  4.802476e+06         2    19574      2  39.561855   
2 -1.352722e+07  4.803311e+06         3    19574      3  39.567636   
3 -1.352789e+07  4.804691e+06         4    19574      4  39.577194   
4 -1.352617e+07  4.796487e+06         5    19610      1  39.520367   

        LONDD  SEGPTID  
0 -121.514390   195741  
1 -121.514891   195742  
2 -121.517107   195743  
3 -121.523048   195744  
4 -121.507617   196101

In [64]:
df3.tail()

X             Y  OBJECTID  SEGMENT  POINT      LATDD  \
139 -1.349062e+07  4.709715e+06       140    62819      3  38.916476   
140 -1.349411e+07  4.719130e+06       141    62977      1  38.982249   
141 -1.349157e+07  4.717313e+06       142    62977      3  38.969563   
142 -1.348944e+07  4.717192e+06       143    62977      4  38.968717   
143 -1.349248e+07  4.718092e+06       144    62977      2  38.975000   

          LONDD  SEGPTID  
139 -121.188276   628193  
140 -121.219620   629771  
141 -121.196789   629773  
142 -121.177650   629774  
143 -121.205000   629772

In [65]:
df4.head()

id                    name  gender      dates  yearOfBirth  yearOfDeath  \
0  10093  Abakanowicz, Magdalena  Female  born 1930       1930.0          NaN   
1      0     Abbey, Edwin Austin    Male  1852–1911       1852.0       1911.0   
2   2756        Abbott, Berenice  Female  1898–1991       1898.0       1991.0   
3      1  Abbott, Lemuel Francis    Male  1760–1803       1760.0       1803.0   
4    622          Abrahams, Ivor    Male  born 1935       1935.0          NaN   

                     placeOfBirth            placeOfDeath  \
0                          Polska                     NaN   
1     Philadelphia, United States  London, United Kingdom   
2      Springfield, United States   Monson, United States   
3  Leicestershire, United Kingdom  London, United Kingdom   
4           Wigan, United Kingdom                     NaN   

                                                 url  
0  http://www.tate.org.uk/art/artists/magdalena-a...  
1  http://www.tate.org.uk/art/artists/edwin-austi...  
2  http://www.tate.org.uk/art/artists/berenice-ab...  
3  http://www.tate.org.uk/art/artists/lemuel-fran...  
4  http://www.tate.org.uk/art/artists/ivor-abraha...

In [66]:
df4.tail()

id                   name gender      dates  yearOfBirth  \
3527  12542        Zorio, Gilberto   Male  born 1944       1944.0   
3528   2186             Zox, Larry   Male  born 1936       1936.0   
3529    621  Zuccarelli, Francesco   Male  1702–1788       1702.0   
3530   2187       Zuloaga, Ignacio   Male  1870–1945       1870.0   
3531   2188        Zyw, Aleksander   Male  1905–1995       1905.0   

      yearOfDeath               placeOfBirth placeOfDeath  \
3527          NaN      Andorno Micca, Italia          NaN   
3528          NaN  Des Moines, United States          NaN   
3529       1788.0                     Italia      Firenze   
3530       1945.0                     España       España   
3531       1995.0                     Polska          NaN   

                                                    url  
3527  http://www.tate.org.uk/art/artists/gilberto-zo...  
3528  http://www.tate.org.uk/art/artists/larry-zox-2186  
3529  http://www.tate.org.uk/art/artists/francesco-z...  
3530  http://www.tate.org.uk/art/artists/ignacio-zul...  
3531  http://www.tate.org.uk/art/artists/aleksander-...

The .head() and .tail() methods return the first and last 5 rows of a dataframe by default, respectively. An integer can be passed into the method in order to return an alternate desired number of rows from the dataframe. 